addons/crm/tests/test_crm_lead_assignment.py

```diff
@@ -459,6 +459,36 @@ def test_assign_quota(self):
             "Assignment quota: no compensation and no limit anymore (do as asked even with 30 already assigned)"
         )
 
+    # test to mitigate random.choice from raising the zero weight validation error
+    def test_random_choice(self):
+        # create fourth sales team with one member
+        sales_team_4 = self.env['crm.team'].create({
+            'name': 'Sales Team 4',
+            'sequence': 15,
+            'alias_name': False,
+            'use_leads': True,
+            'use_opportunities': True,
+            'company_id': False,
+            'user_id': False,
+            'assignment_domain': [('country_id', '!=', False)],
+        })
+
+        sales_team_4_m1 = self.env['crm.team.member'].create({
+            'user_id': self.user_sales_manager.id,
+            'crm_team_id': sales_team_4.id,
+            'assignment_max': 30,
+            'assignment_domain': False,
+        })
+
+        # assuming that the team member was manually assigned 500 leads in the span of the last 30 days
+        sales_team_4_m1.lead_month_count = 500
```

Comment:
You should just set that when creating the team member

Reply:
I need to hardcode the lead count for the team member here because the field is otherwise computed to 0 as per https://github.com/odoo/odoo/blob/047512e89163eabc03c3f144c9c46cf0292a7a55/addons/crm/models/crm_team_member.py#L31 (Because we're creating a brand new member for this test case)